CDEPS-interface/CMakeLists.txt

```diff
@@ -2,13 +2,13 @@
 if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
   set(CMAKE_Fortran_FLAGS "-g -fbacktrace -ffree-line-length-none")
   if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
-    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz")
+    set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch -fallow-invalid-boz -DCPRGNU")
   endif()
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
   set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -fcheck=bounds -ffpe-trap=invalid,zero,overflow,underflow" )
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
   set(CMAKE_Fortran_FLAGS "-g -traceback")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O -assume realloc_lhs")
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -O -assume realloc_lhs -DCPRINTEL")
```

Comment:
@ulmononian 
If I could perhaps make a recommendation here.
Identify which objects need this pre-processor directive and then use `target_compile_definitions` just on that`target`.

Reply:
i attempted this alternative by adding the following section after line 26 here: https://github.com/ufs-community/ufs-weather-model/blob/dd41cc61e5e52e2cfeb00af8930e7496a5fb91fd/CDEPS-interface/CMakeLists.txt#L21-L26: 

```
if(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
  target_compile_definitions(CPRINTEL)
elseif(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
  if(${CMAKE_Fortran_COMPILER_VERSION} VERSION_GREATER_EQUAL 10)
    target_compile_definitions(CPRGNU)
  endif()
endif()
```
but it did not work for me. 

also, i am not sure how to add something equivalent to this effect to the [CMEPS CMakelists.txt ](https://github.com/ufs-community/ufs-weather-model/blob/develop/CMEPS-interface/CMakeLists.txt), as it only references the CDEPS/ufs/cdeps_share F90 files here https://github.com/ufs-community/ufs-weather-model/blob/dd41cc61e5e52e2cfeb00af8930e7496a5fb91fd/CMEPS-interface/CMakeLists.txt#L41-L42.